server/kolide/osquery_options.go

```diff
@@ -0,0 +1,41 @@
+package kolide
+
+import "encoding/json"
+
+type OsqueryOptionsStore interface {
+	ApplyOptions(options *OptionsSpec) error
+	GetOptions() (*OptionsSpec, error)
+	OptionsForHost(host *Host) (json.RawMessage, error)
```

Comment:
The idea of `json.RawMessage` here is to support any osquery configuration without explicitly having to mention it in an OsqueryConfig object. This might be cleaner and achieve the same effect if `json.RawMessage` were replaced with `map[string]interface{}`.